wgpu-native/src/command/transfer.rs

```diff
@@ -146,8 +146,10 @@ pub fn command_encoder_copy_buffer_to_texture<B: GfxBackend>(
     let hub = B::hub();
     let mut token = Token::root();
 
+    let (device_guard, mut token) = hub.devices.read(&mut token);
     let (mut cmb_guard, mut token) = hub.command_buffers.write(&mut token);
     let cmb = &mut cmb_guard[command_encoder_id];
+    let features = &device_guard[cmb.device_id.value].features;
```

Comment:
let's add a TODO here to store the features in the command buffer to begin with and avoid locking the device each time

Reply:
Yeah I was wondering about this :)

I can probably just copy whichever features we need into the command buffer to start with. So maybe we should have `DeviceFeatures` and a subset of these called `CommandBufferFeatures`?

Reply:
Perhaps that's the end picture for us. Today we can just copy the whole thing :)

Reply:
Agreed, I'll make it `Copy` for now